zephyr/Kconfig

```diff
@@ -12,12 +12,26 @@ config WPA_SUPP
     select NET_SOCKETS_PACKET
     select NET_SOCKETPAIR
     select NET_L2_WIFI_MGMT
+    # For newlib min HEAP check, NCS_SAMPLE_DEFAULTS doesn't print assert information
+    select LOG_DEFAULT_MINIMAL
+	select ASSERT
+    select ASSERT_VERBOSE
+    # Without this reset on fatal error is triggered
+    select DEBUG
+    ###
     select EXPERIMENTAL if !SOC_NRF5340_CPUAPP_QKAA
     help
       WPA supplicant implements 802.1X related functions.
 
 if WPA_SUPP
 
+# For asserts, this avoids continuous reset
```

Comment:
Should this be guarded by `CONFIG_ASSERT`? So that it doesn't become the default case when asserts are disabled, which will be the common case for most applications.

Reply:
But we are selecting `ASSERT` above unconditionally, so, it would be redundant.

Reply:
I understand ASSERTS are not always desired but as of now that is the only to catch this low HEAP scenario gracefully.